pkg/pillar/cmd/baseosmgr/baseosmgr.go

```diff
@@ -515,3 +541,44 @@ func handleZbootConfigDelete(ctxArg interface{}, key string,
 	// in any case
 	log.Functionf("handleZbootConfigDelete(%s) done", key)
 }
+
+// returns retryUpdateCounter if the file exists
+// else returns 0
+func readSavedRetryUpdateCounter(fromConfig bool) uint32 {
+	fileName := currentRetryUpdateCounterFile
+	if fromConfig {
+		fileName = cfgRetryUpdateCounterFile
+	}
+	log.Tracef("readSavedRetryUpdateCounter - reading %s", fileName)
+
+	b, err := ioutil.ReadFile(fileName)
+	if err == nil {
+		c, err := strconv.Atoi(string(b))
+		if err != nil {
+			log.Errorf("readSavedRetryUpdateCounter: %s", err)
+		}
+		return uint32(c)
+	}
+	log.Warnf("readSavedRetryUpdateCounter - %s doesn't exist", fileName)
+	return 0
+}
+
+func saveRetryUpdateCounter(fromConfig bool, retryUpdateCounter uint32) {
```

Comment:
This fromConfig bool doesn't make the calling code easy to read and understand.
Either define two distinct enum values or two separate wrapper functions (which can pass the filename to an Impl function).

Reply:
I splited it into separate functions